### PR TITLE
[MM-24764] [MM-24673] Dont call member counts by group when no license or permission

### PR DESCRIPTION
**Pull request by fmunshi**
#### Summary
- Ensures that `getChannelMemberCountsByGroup` is called only when the server has license for LDAP & when the user has permission
- Adds `getChannelMemberCountsByGroup` to component mount on `create_post` component

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-24764 (License and permission present before calling member counts action)
- https://mattermost.atlassian.net/browse/MM-24673 (Ensure member counts action dispatched on create post component mounted)

### CODE REVIEW

**Comment by hahmadia** [comment]
Hey Farhan! Should we add `getChannelMemberCountsByGroup` to component mount of `create_comment` as well?

File: actions/websocket_actions.jsx [hunk 1]
```diff
@@ -737,10 +738,8 @@ function handleUserAddedEvent(msg) {
             type: UserTypes.RECEIVED_PROFILE_IN_CHANNEL,
             data: {id: msg.broadcast.channel_id, user_id: msg.data.user_id},
         });
-        if (isTimezoneEnabled) {
-            dispatch(getChannelMemberCountsByGroup(currentChannelId, true));
-        } else {
-            dispatch(getChannelMemberCountsByGroup(currentChannelId, false));
+        if (license && license.IsLicensed === 'true' && license.LDAPGroups === 'true') {
```
Comment (hahmadia) [inline actions/websocket_actions.jsx:741]:
```suggestion
        if (license?.IsLicensed === 'true' && license?.LDAPGroups === 'true') {
```

File: actions/websocket_actions.jsx [hunk 2]
```diff
@@ -796,10 +796,8 @@ export async function handleUserRemovedEvent(msg) {
             type: UserTypes.RECEIVED_PROFILE_NOT_IN_CHANNEL,
             data: {id: msg.broadcast.channel_id, user_id: msg.data.user_id},
         });
-        if (isTimezoneEnabled) {
-            dispatch(getChannelMemberCountsByGroup(currentChannel.id, true));
-        } else {
-            dispatch(getChannelMemberCountsByGroup(currentChannel.id, false));
+        if (license && license.IsLicensed === 'true' && license.LDAPGroups === 'true') {
```
Comment (hahmadia) [inline actions/websocket_actions.jsx:799]:
```suggestion
        if (license?.IsLicensed === 'true' && license?.LDAPGroups === 'true') {
```

File: actions/websocket_actions.jsx [hunk 3]
```diff
@@ -845,9 +843,13 @@ export async function handleUserUpdatedEvent(msg) {
     const user = msg.data.user;
 
     const config = getConfig(state);
-    const isTimezoneEnabled = config.ExperimentalTimezone === 'true';
+    const license = getLicense(state);
+
     const userIsGuest = isGuest(user);
-    if (userIsGuest || isTimezoneEnabled) {
+    const isTimezoneEnabled = config.ExperimentalTimezone === 'true';
+    const isLDAPEnabled = license && license.IsLicensed === 'true' && license.LDAPGroups === 'true';
```
Comment (hahmadia) [inline actions/websocket_actions.jsx:850]:
```suggestion
    const isLDAPEnabled = license?.IsLicensed === 'true' && license?.LDAPGroups === 'true';
```

File: components/create_comment/index.js [hunk 4]
```diff
@@ -3,14 +3,14 @@
...
-import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {getConfig, getLicense} from 'mattermost-redux/selectors/entities/general';
 import {isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
 import {haveIChannelPermission} from 'mattermost-redux/selectors/entities/roles';
 import {getBool} from 'mattermost-redux/selectors/entities/preferences';
 import {getAllChannelStats, getChannelMemberCountsByGroup} from 'mattermost-redux/selectors/entities/channels';
 import {makeGetMessageInHistoryItem} from 'mattermost-redux/selectors/entities/posts';
 import {resetCreatePostRequest, resetHistoryIndex} from 'mattermost-redux/actions/posts';
-import {getChannelTimezones, getChannelMemberCountsByGroup as selectChannelMemberCountsByGroup} from 'mattermost-redux/actions/channels';
+import {getChannelTimezones, getChannelMemberCountsByGroup as loadChannelMemberCountsByGroup} from 'mattermost-redux/actions/channels';
```
Comment (hahmadia) [inline components/create_comment/index.js:13]: Actually, to keep this somewhat synchronized with mobile, it as suggested to me that keep this called as `getChannelMemberCountsByGroup` and import the selector above `as selectChannelMemberCountsByGroup`. 0/5 but just idea to keep it parallel with mobile code but not super important. I'll leave it up to cause currently it's good too.

File: components/create_comment/index.js [hunk 5]
```diff
@@ -68,7 +69,8 @@ function makeMapStateToProps() {
             team: channel.team_id,
             permission: Permissions.USE_CHANNEL_MENTIONS,
         });
-        const useGroupMentions = haveIChannelPermission(state, {
+        const isLDAPEnabled = license && license.IsLicensed === 'true' && license.LDAPGroups === 'true';
```
Comment (hahmadia) [inline components/create_comment/index.js:72]:
```suggestion
        const isLDAPEnabled = license?.IsLicensed === 'true' && license?.LDAPGroups === 'true';
```

File: components/create_post/index.js [hunk 6]
```diff
@@ -84,7 +85,8 @@ function makeMapStateToProps() {
             team: currentChannel.team_id,
             permission: Permissions.USE_CHANNEL_MENTIONS,
         });
-        const useGroupMentions = haveIChannelPermission(state, {
+        const isLDAPEnabled = license && license.IsLicensed === 'true' && license.LDAPGroups === 'true';
```
Comment (hahmadia) [inline components/create_post/index.js:88]:
```suggestion
        const isLDAPEnabled = license?.IsLicensed === 'true' && license?.LDAPGroups === 'true';
```

File: components/create_comment/index.js (hunk 4 above)
Comment (fmunshi) [reply components/create_comment/index.js]: done @hahmadia

**Pull request merged**